### DISCUSSION TITLE
How to prevent load data on initial load?

### DISCUSSION THREAD

**Discussion by SohelIslamImran** [algolia/instantsearch#5625] [Q&A] [answered]
How to prevent load data on initial load in TS?

**Comment by dhayab** [comment]
Hi, you can prevent empty search requests from returning results by modifiying the search client as documented here: https://www.algolia.com/doc/guides/building-search-ui/going-further/conditional-requests/js/#detecting-empty-search-requests.

**Reply by SohelIslamImran** [reply]
@dhayab  But it throws so many typescript errors. The type and properties of `algoliaClient` are read-only. There should be an easy way to do this by configuration or options!

![image](https://github.com/algolia/instantsearch/assets/76745339/65892a6d-1331-4f5a-b4d0-3bf8d37ba0a5)

**Reply by dhayab** [reply, marked as answer]
This code throws type errors because the results you're returning are missing a number of other expected keys. It should look like this, as shown in the guide:

```ts
const algoliaSearchClient: SearchClient = {
  ...algoliaClient,
  search(requests, requestOptions) {
    if (requests.every(({ params }) => !params.query)) {
      return Promise.resolve({
        results: requests.map(() => ({
          hits: [],
          nbHits: 0,
          nbPages: 0,
          page: 0,
          processingTimeMS: 0,
          hitsPerPage: 0,
          exhaustiveNbHits: false,
          query: "",
          params: ""
        }))
      });
    }

    return algoliaClient.search(requests, requestOptions);
  }
};
```

**Reply by SohelIslamImran** [reply]
@dhayab Thank you so much. It works. But still, there should be an easy way to do this by configuration or options! 
Like :
```js
const algoliaClient = algoliasearch(
  'YourApplicationID',
  'YourSearchOnlyAPIKey', 
  { searchOnInit: false }
);
```

**Reply by dhayab** [reply]
Thanks for your feedback. I'll forward it to the team, to see if and how we can prioritize it.
